docker-compose.yaml

```diff
@@ -0,0 +1,180 @@
+version: '3'
+
+# This docker-compose is for developer convenience, not for running in production.
+
+services:
+
+  postgres:  # for hive
+    image: postgres:16.3
+    ports:
+      - 5432:5432
+    environment:
+      POSTGRES_USER: hive
+      POSTGRES_PASSWORD: hivepassword
+      POSTGRES_DB: hive
+
+  kafka:
+    image: apache/kafka-native:4.0.0
+    ports:
+      - 9092:9092
+    environment:
+      # default config
+      KAFKA_NODE_ID: 1
+      KAFKA_PROCESS_ROLES: broker,controller
+      KAFKA_LISTENERS: PLAINTEXT://:9092,CONTROLLER://localhost:9093
+      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://kafka:9092
+      KAFKA_CONTROLLER_LISTENER_NAMES: CONTROLLER
+      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT
+      KAFKA_CONTROLLER_QUORUM_VOTERS: 1@localhost:9093
+      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+      KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR: 1
+      KAFKA_TRANSACTION_STATE_LOG_MIN_ISR: 1
+      KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS: 0
+      KAFKA_NUM_PARTITIONS: 3
+      
+      # custom config
+      KAFKA_DELETE_TOPIC_ENABLE: true
+
+  minio:
+    image: minio/minio:RELEASE.2025-02-07T23-21-09Z
+    environment:
+      MINIO_ROOT_USER: miniouser
+      MINIO_ROOT_PASSWORD: miniopassword
+    ports:
+      - 9000:9000
+      - 9001:9001
+    command: 'server /data --console-address ":9001"'
+
+  minio-create-bucket:
+    image: minio/mc
+    depends_on:
+      - minio
+    environment:
+      MINIO_URL: http://minio:9000
+      MINIO_USER: miniouser
+      MINIO_PWD: miniopassword
+      MINIO_CTS_USER: cts
+      MINIO_CTS_PWD: ctspassword
+    entrypoint: /s3_policies/minio_create_bucket_entrypoint.sh
+    volumes:
+      - ./s3_policies/cdm-task-service-read-write-policy.json:/s3_policies/cdm-task-service-read-write-policy.json
+      - ./s3_policies/minio_create_bucket_entrypoint.sh:/s3_policies/minio_create_bucket_entrypoint.sh
+
+  mongodb:
+    image: mongo:7.0.14
+    ports:
+      - 27017:27017
+    # environment:
+    #  - MONGO_INITDB_ROOT_USERNAME=root
+    #  - MONGO_INITDB_ROOT_PASSWORD=secret
+
+  auth2:
```

Comment:
can you remind me what does local auth server do?

Reply:
We're going to have to config the event processor with a CTS admin server token so it can pull any job. If we run the server locally we can create those tokens on the fly rather than have to use real developer tokens for local testing as well as in GHA.